csharp/src/Ice/LocatorInfo.cs

```diff
@@ -525,26 +626,20 @@ private static void TraceWellKnown(string msg, Reference wellKnown, Reference in
                 {
                     lock (_mutex)
                     {
-                        _wellKnownProxyRequests.Remove(reference);
+                        _wellKnownProxyRequests.Remove((reference.Identity, reference.Protocol));
                     }
                 }
             }
         }
 
-        private Reference? RemoveLocation(Reference reference) =>
-            _locationCache.TryRemove(reference, out (TimeSpan InsertionTime, Reference Reference) entry) ?
-                entry.Reference : null;
-
-        private Reference? RemoveWellKnownProxy(Reference reference) =>
-            _wellKnownProxyCache.TryRemove(reference,
-                out (TimeSpan InsertionTime, Reference Reference) entry) ? entry.Reference : null;
-
-        private sealed class LocationComparer : IEqualityComparer<Reference>
+        private sealed class LocationComparer : IEqualityComparer<(Location Location, Protocol Protocol)>
         {
-            public bool Equals(Reference? x, Reference? y) =>
-                x!.Location.SequenceEqual(y!.Location) && x.Protocol == y.Protocol;
+            public bool Equals(
+                (Location Location, Protocol Protocol) x,
+                (Location Location, Protocol Protocol) y) =>
```

Comment:
Minor but we usually use lhs/rhs not x/y

Reply:
Fixed. I used the IEqualityComparer parameter names: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.iequalitycomparer-1.equals?view=net-5.0